beets/autotag/mb.py

```diff
@@ -210,9 +210,18 @@ def track_info(recording, index=None, medium=None, medium_index=None,
     lyricist = []
     composer = []
     composer_sort = []
+    work = []
+    work_id = []
+    work_disambig = []
     for work_relation in recording.get('work-relation-list', ()):
         if work_relation['type'] != 'performance':
             continue
+        work.append(work_relation['work']['title'])
+        work_id.append(work_relation['work']['id'])
+        if 'disambiguation' in work_relation['work']:
+            work_disambig.append(work_relation['work']['disambiguation'])
+        else:
+            work_disambig.append('')
```

Comment:
Is this necessary? Seems like an empty list, in the case of no disambiguation, would be fine…

Reply:
The point is that if there are several works, some have disambiguations and some don't, we want to keep track of which work has which disambiguation. I's a rather rare case, but in classical music, the name of the arrangement typically is in the disambiguation. Usually it is in the disambiguation of all works though. Maybe it would be a better thing to just integrate the disambiguation into the work title, but that might be too much... Actually, beets is not fetching the disambiguations for a while lot of things, first among them the artists/release artists, is there a reason for that?

Reply:
I see! So in the case of a missing disambiguation string for a work, will we get something like `foo, , bar` in the final disambiguation string? I guess that's fine, but it looks a little weird…

> Actually, beets is not fetching the disambiguations for a while lot of things, first among them the artists/release artists, is there a reason for that?

The answer is pretty simple: nobody has ever asked for them. :smiley:

Reply:
Yes it's exactly that, and that's the best I could come up with... For works I feel that the disambiguations are needed, for example to distinguish all the different arrangements (even if one could argue that the 'arranger' tag is there for that, but then we would need an 'arranger_sort' tag) or the successive versions of a work (Mahler's symphonies or Bach's Passions have several versions, some works have versions for two pianos and others for orchestra or piano quintet like the piano quintet of Brahms). I just feel that for MB entities, the name, sort name and disambiguation belong together and fetching one without the others feels weird (especially name and sort name). 
Maybe we could assume that if a disambiguation affects one work, it will logically affect all the other works in the recording (as would be the case in all the examples I can think of) and don't need the sort of dirty meddling I did before. Maybe to make it even cleaner (but the code gets dirtier) we can organize it that there is only one disambiguation, we keep only the first disambiguation from the first work that has one.